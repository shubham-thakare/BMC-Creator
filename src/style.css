@import "~@blueprintjs/core/lib/css/blueprint.css";
@import "~@blueprintjs/icons/lib/css/blueprint-icons.css";

:root {
  --app-background-color: #f5f8fa;
  --app-divider-color: #182026;
  --link-color: #182026;
  --link-hover-color: #818181;
  --white-color: #ffffff;
  --black-color: #000000;
  --red-color: #ff0000;
  --title-input-color: #000000;
  --title-input-placeholder-color: #8a9ba8;
  --placeholder-color: #8a9ba8;
  --title-input-font-size: 32px;
  --input-font-size: 14px;
  --navbar-link-font-size: 18px;
  --navbar-link-font-size-smaller: 14px;
  --input-font-family: -apple-system, "BlinkMacSystemFont", "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Open Sans", "Helvetica Neue", "Icons16", sans-serif;
  --spacing-scale-1: 8px;
  --elevation-1: 0px 0px 5px -2px;
  --elevation-2: -1px 2px 5px 0px rgba(0, 0, 0, 0.2), 2px 2px 2px 0 rgba(0, 0, 0, 0.19);
}

* {
  box-sizing: border-box;
}

body {
  font-family: -apple-system, "BlinkMacSystemFont", "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Cantarell", "Open Sans", "Helvetica Neue", "Icons16", sans-serif;
  overflow-x: hidden;
  margin: 0;
  background: var(--app-background-color);
  user-select: none;
  -webkit-user-drag: none;
}

/* width */
::-webkit-scrollbar {
  width: 10px;
}

/* Track */
::-webkit-scrollbar-track {
  background: transparent;
}

/* Handle */
::-webkit-scrollbar-thumb {
  background: rgba(0, 0, 0, 0.25);
  -webkit-border-radius: 10px;
  border-radius: 10px;
}

/* Handle on hover */
::-webkit-scrollbar-thumb:hover {
  background: rgba(0, 0, 0, 0.35);
}

span:focus {
  outline: none;
}

.bp3-editable-text.bp3-editable-text-placeholder:hover, .bp3-editable-text.bp3-editable-text-editing:hover {
  box-shadow: 0 0 0 0 rgba(19,124,189,0), 0 0 0 0 rgba(19,124,189,0), inset 0 0 0 1px rgba(16,22,26,0.15);
}

.bp3-editable-text.bp3-editable-text-placeholder:before, .bp3-editable-text.bp3-editable-text-editing:before {
  box-shadow: none;
}

.bp3-editable-text.bp3-editable-text-placeholder, .bp3-editable-text.bp3-editable-text-editing {
  padding: 5px;
  border-radius: 5px;
}

button.bp3-button {
  outline: none;
}

.full-width {
  width: 100%;
}

.text-bold {
  font-weight: bold;
}

.text-italic {
  font-style: italic;
}

.no-animation {
  animation: none !important;
}