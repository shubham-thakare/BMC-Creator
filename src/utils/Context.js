import { createContext } from 'react';
import htmlToImage from 'html-to-image';
import wait from 'waait';
import cmsData from '../cms';

export const AppContext = createContext({});

const filter = (node) => {
  return node.tagName !== 'i';
};

const exportImage = async () => {
  hideNav();
  await wait(500);
  const node = document.getElementById('canvas-screen');
  closeNav();
  htmlToImage.toSvgDataURL(node, { filter: filter }).then((dataUrl) => {
    var link = document.createElement('a');
    link.download = 'my-image-name.svg';
    link.href = dataUrl;
    link.click();
  });
};

const hideNav = () => {
  document.getElementById('sidebar').style.left = '-220px';
  document.getElementById('canvas-screen').style.paddingLeft = '10px';
};

const openNav = () => {
  document.getElementById('sidebar').style.left = '0';
  document.getElementById('canvas-screen').style.paddingLeft = '230px';
  document.getElementById('open-button').style.display = 'none';
  document.getElementById('toggle-hide').style.display = 'block';
};

const closeNav = () => {
  document.getElementById('sidebar').style.left = '-170px';
  document.getElementById('canvas-screen').style.paddingLeft = '60px';
  document.getElementById('open-button').style.display = 'block';
  document.getElementById('toggle-hide').style.display = 'none';
};

export const contextWithoutState = {
  cmsData,
  exportImage,
  openNav,
  closeNav,
};
