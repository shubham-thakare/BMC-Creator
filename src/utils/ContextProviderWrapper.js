import React, { useReducer } from 'react';
import PropTypes from 'prop-types';
import { ContextProvider } from '.';
import { reducer, appState } from '../reducer';
import { mockState } from '../mock';

export const ContextProviderWrapper = ({ children, useMockState }) => {
  const [state, dispatch] = useReducer(
    reducer,
    useMockState ? mockState : appState,
  );

  return (
    <ContextProvider value={{ state, dispatch }}>{children}</ContextProvider>
  );
};

ContextProviderWrapper.propTypes = {
  useMockState: PropTypes.bool,
};
