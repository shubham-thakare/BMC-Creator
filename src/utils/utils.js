import { createContext } from 'react';
import htmlToImage from 'html-to-image';
import wait from 'waait';
import { UPDATE_STATE_FROM_FILE_DATA } from '../actions';

export const AppContext = createContext({});
export const ContextProvider = AppContext.Provider;
export const ContextConsumer = AppContext.Consumer;

const filter = (node) => {
  return node.tagName !== 'i';
};

export const exportImage = async (fileName) => {
  hideNav();
  await wait(500);
  const node = document.getElementById('canvas-screen');
  closeNav();
  htmlToImage.toSvgDataURL(node, { filter: filter }).then((dataUrl) => {
    const link = document.createElement('a');
    link.download = fileName + '.svg';
    link.href = dataUrl;
    link.click();
  });
};

export const hideNav = () => {
  try {
    document.getElementById('sidebar').style.left = '-220px';
    document.getElementById('canvas-screen').style.paddingLeft = '10px';
  } catch (ex) {}
};

export const openNav = () => {
  try {
    document.getElementById('sidebar').style.left = '0';
    document.getElementById('canvas-screen').style.paddingLeft = '230px';
    document.getElementById('open-button').style.display = 'none';
    document.getElementById('toggle-hide').style.display = 'block';
  } catch (ex) {}
};

export const closeNav = () => {
  try {
    document.getElementById('sidebar').style.left = '-170px';
    document.getElementById('canvas-screen').style.paddingLeft = '60px';
    document.getElementById('open-button').style.display = 'block';
    document.getElementById('toggle-hide').style.display = 'none';
  } catch (ex) {}
};

export const saveFile = (currentState, fileName) => {
  try {
    const pom = document.createElement('a');
    const blob = new Blob([JSON.stringify(currentState)], { type: 'text/xml' });
    pom.setAttribute('href', window.URL.createObjectURL(blob));
    pom.setAttribute('download', fileName + '.bmc');
    pom.dataset.downloadurl = ['text/xml', pom.download, pom.href].join(':');
    pom.draggable = true;
    pom.classList.add('dragout');
    pom.click();
  } catch (ex) {}
};

export const openFile = (dispatch) => {
  try {
    const inputElement = document.createElement('input');
    inputElement.type = 'file';
    inputElement.accept = '.bmc';
    inputElement.addEventListener('change', (event) => {
      const input = event.target;
      const reader = new FileReader();
      reader.onload = () => {
        const data = reader.result;
        dispatch({ action: UPDATE_STATE_FROM_FILE_DATA, payload: data });
      };
      reader.readAsText(input.files[0]);
    });
    inputElement.dispatchEvent(new MouseEvent('click'));
  } catch (ex) {}
};

export const trimString = (string) => {
  return string ? string.toString().trim() : '';
};

export const userVisitedBefore = () => {
  const userVisitedBefore = getCookie('userVisitedBefore');
  setCookie('userVisitedBefore', true, 365);

  return userVisitedBefore;
};

const setCookie = (cname, cvalue, exdays) => {
  const d = new Date();
  d.setTime(d.getTime() + exdays * 24 * 60 * 60 * 1000);
  const expires = 'expires=' + d.toUTCString();
  document.cookie = cname + '=' + cvalue + ';' + expires + ';path=/';
};

const getCookie = (cname) => {
  const name = cname + '=';
  const ca = document.cookie.split(';');
  for (let i = 0; i < ca.length; i++) {
    let c = ca[i];
    while (c.charAt(0) === ' ') {
      c = c.substring(1);
    }
    if (c.indexOf(name) === 0) {
      return c.substring(name.length, c.length);
    }
  }
  return '';
};
