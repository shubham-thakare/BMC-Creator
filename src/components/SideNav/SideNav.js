/* eslint-disable no-script-url */
/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Icon, Divider } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import {
  SideNavbar,
  AppTitle,
  Link,
  ToggleHide,
  Footer,
} from './styles/SideNav.style';
import { AppContext } from '../../utils';
import { OPEN_NAV, CLOSE_NAV, EXPORT_IMAGE } from '../../actions';
import cmsData from '../../cms';
import { SAVE_FILE, OPEN_FILE } from '../../actions/actions';

const SideNav = ({
  hasSaveFileOption,
  hasOpenFileOption,
  hasPrintOption,
  hasExportOption,
  hasPresentationModeOption,
  hasFooter,
  ...props
}) => {
  const { dispatch } = useContext(AppContext);
  const {
    l_app_title,
    l_app_version,
    l_save,
    l_open,
    l_export_to_svg,
    l_presentation_mode,
    l_view_on_github,
    url_app_github,
  } = cmsData;

  return (
    <>
      <SideNavbar id="sidebar" {...props}>
        <Link
          href="javascript:void(0)"
          id="open-button"
          className="openbtn joyride-first-step"
          onClick={() => dispatch({ action: OPEN_NAV })}
        >
          <Icon icon={IconNames.MENU} iconSize={30} />
        </Link>
        <Link
          href="javascript:void(0)"
          id="close-button"
          className="closebtn"
          onClick={() => dispatch({ action: CLOSE_NAV })}
        >
          <Icon icon={IconNames.CHEVRON_LEFT} iconSize={30} />
        </Link>

        <ToggleHide id="toggle-hide">
          <Divider />
          <AppTitle>
            <b>{l_app_title}</b> <i>{l_app_version}</i>
          </AppTitle>
          <Divider />
          {hasSaveFileOption && (
            <Link
              href="javascript:void(0)"
              onClick={() => dispatch({ action: SAVE_FILE })}
            >
              <Icon icon={IconNames.DOCUMENT} iconSize={20} /> {l_save}
            </Link>
          )}
          {hasOpenFileOption && (
            <Link
              href="javascript:void(0)"
              onClick={() =>
                dispatch({ action: OPEN_FILE, payload: { callback: dispatch } })
              }
            >
              <Icon icon={IconNames.DOCUMENT_OPEN} iconSize={20} /> {l_open}
            </Link>
          )}
          {hasPrintOption && (
            <Link href="javascript:void(0)" onClick={() => window.print()}>
              <Icon icon={IconNames.PRINT} iconSize={20} /> Print
            </Link>
          )}
          {hasExportOption && (
            <Link
              href="javascript:void(0)"
              onClick={() => dispatch({ action: EXPORT_IMAGE })}
            >
              <Icon icon={IconNames.EXPORT} iconSize={20} /> {l_export_to_svg}
            </Link>
          )}
          {hasPresentationModeOption && (
            <Link href="javascript:void(0)">
              <Icon icon={IconNames.PRESENTATION} iconSize={20} />{' '}
              {l_presentation_mode}
            </Link>
          )}

          {hasFooter && (
            <Footer>
              <Divider />
              <Link
                href={url_app_github}
                target="_blank"
                rel="noopener noreferrer"
              >
                {l_view_on_github} <Icon icon={IconNames.SHARE} iconSize={14} />
              </Link>
            </Footer>
          )}
        </ToggleHide>
      </SideNavbar>
    </>
  );
};

SideNav.propTypes = {
  hasSaveFileOption: PropTypes.bool,
  hasOpenFileOption: PropTypes.bool,
  hasPrintOption: PropTypes.bool,
  hasExportOption: PropTypes.bool,
  hasPresentationModeOption: PropTypes.bool,
  hasFooter: PropTypes.bool,
};

export default SideNav;
