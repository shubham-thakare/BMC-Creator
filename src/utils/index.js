import { AppContext, contextWithoutState } from './Context';

export { AppContext, contextWithoutState };
