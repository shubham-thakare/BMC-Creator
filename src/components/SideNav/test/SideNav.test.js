import React from 'react';
import { mount } from 'enzyme';
import { Divider } from '@blueprintjs/core';
import SideNav from '..';
import { ContextProviderWrapper } from '../../../utils';
import { Link, ToggleHide, AppTitle, Footer } from '../styles/SideNav.style';

let wrapper;

const props = {
  hasSaveFileOption: true,
  hasOpenFileOption: true,
  hasPrintOption: true,
  hasExportOption: true,
  hasPresentationModeOption: true,
  hasFooter: true,
};

describe('SideNav Component', () => {
  beforeAll(() => {
    wrapper = mount(
      <ContextProviderWrapper useMockState>
        <SideNav {...props} />
      </ContextProviderWrapper>,
    );
  });

  it('To match snapshot of SideNav', async (done) => {
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('Should have 8 links', async (done) => {
    expect(wrapper.find(Link)).toHaveLength(8);
    done();
  });

  it('Should have 1 ToggleHide component', async (done) => {
    expect(wrapper.find(ToggleHide)).toHaveLength(1);
    done();
  });

  it('Should have 1 AppTitle component', async (done) => {
    expect(wrapper.find(AppTitle)).toHaveLength(1);
    done();
  });

  describe('Footer Component', () => {
    it('Should have 1 Divider component', async (done) => {
      expect(wrapper.find(Footer).find(Divider)).toHaveLength(1);
      done();
    });
    it('Should have 1 Link', async (done) => {
      expect(wrapper.find(Footer).find(Link)).toHaveLength(1);
      done();
    });
  });

  it('Simulate Link clicks', async (done) => {
    wrapper
      .find(Link)
      .at(0)
      .simulate('click');
    wrapper
      .find(Link)
      .at(1)
      .simulate('click');
    wrapper
      .find(Link)
      .at(2)
      .simulate('click');
    wrapper
      .find(Link)
      .at(3)
      .simulate('click');
    wrapper
      .find(Link)
      .at(4)
      .simulate('click');
    wrapper
      .find(Link)
      .at(5)
      .simulate('click');
    wrapper
      .find(Link)
      .at(6)
      .simulate('click');
    wrapper
      .find(Link)
      .at(7)
      .simulate('click');
    done();
  });
});
