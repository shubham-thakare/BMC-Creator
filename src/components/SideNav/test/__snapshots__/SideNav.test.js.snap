// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNav Component To match snapshot of SideNav 1`] = `
<ContextProviderWrapper
  useMockState={true}
>
  <SideNav
    hasExportOption={true}
    hasFooter={true}
    hasOpenFileOption={true}
    hasPresentationModeOption={true}
    hasPrintOption={true}
    hasSaveFileOption={true}
  >
    <SideNavstyle__SideNavbar
      id="sidebar"
    >
      <div
        className="SideNavstyle__SideNavbar-sc-1s7cd9d-0 gblKmu"
        id="sidebar"
      >
        <SideNavstyle__Link
          className="openbtn joyride-first-step"
          href="javascript:void(0)"
          id="open-button"
          onClick={[Function]}
        >
          <a
            className="SideNavstyle__Link-sc-1s7cd9d-2 jBzRZo openbtn joyride-first-step"
            href="javascript:void(0)"
            id="open-button"
            onClick={[Function]}
          >
            <Blueprint3.Icon
              icon="menu"
              iconSize={30}
            >
              <span
                className="bp3-icon bp3-icon-menu"
                icon="menu"
              >
                <svg
                  data-icon="menu"
                  height={30}
                  viewBox="0 0 20 20"
                  width={30}
                >
                  <desc>
                    menu
                  </desc>
                  <path
                    d="M1 6h18c.55 0 1-.45 1-1s-.45-1-1-1H1c-.55 0-1 .45-1 1s.45 1 1 1zm18 3H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1zm0 5H1c-.55 0-1 .45-1 1s.45 1 1 1h18c.55 0 1-.45 1-1s-.45-1-1-1z"
                    fillRule="evenodd"
                    key="0"
                  />
                </svg>
              </span>
            </Blueprint3.Icon>
          </a>
        </SideNavstyle__Link>
        <SideNavstyle__Link
          className="closebtn"
          href="javascript:void(0)"
          id="close-button"
          onClick={[Function]}
        >
          <a
            className="SideNavstyle__Link-sc-1s7cd9d-2 jBzRZo closebtn"
            href="javascript:void(0)"
            id="close-button"
            onClick={[Function]}
          >
            <Blueprint3.Icon
              icon="chevron-left"
              iconSize={30}
            >
              <span
                className="bp3-icon bp3-icon-chevron-left"
                icon="chevron-left"
              >
                <svg
                  data-icon="chevron-left"
                  height={30}
                  viewBox="0 0 20 20"
                  width={30}
                >
                  <desc>
                    chevron-left
                  </desc>
                  <path
                    d="M8.41 10l5.29-5.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71l-6 6c-.18.18-.29.43-.29.71 0 .28.11.53.29.71l6 6a1.003 1.003 0 001.42-1.42L8.41 10z"
                    fillRule="evenodd"
                    key="0"
                  />
                </svg>
              </span>
            </Blueprint3.Icon>
          </a>
        </SideNavstyle__Link>
        <SideNavstyle__ToggleHide
          id="toggle-hide"
        >
          <div
            className="SideNavstyle__ToggleHide-sc-1s7cd9d-3 hIydNM"
            id="toggle-hide"
          >
            <Blueprint3.Divider>
              <div
                className="bp3-divider"
              />
            </Blueprint3.Divider>
            <SideNavstyle__AppTitle>
              <div
                className="SideNavstyle__AppTitle-sc-1s7cd9d-1 chyrJg"
              >
                <b>
                  Business Model Canvas
                </b>
                 
                <i>
                  v1.0.0
                </i>
              </div>
            </SideNavstyle__AppTitle>
            <Blueprint3.Divider>
              <div
                className="bp3-divider"
              />
            </Blueprint3.Divider>
            <SideNavstyle__Link
              href="javascript:void(0)"
              onClick={[Function]}
            >
              <a
                className="SideNavstyle__Link-sc-1s7cd9d-2 jBzRZo"
                href="javascript:void(0)"
                onClick={[Function]}
              >
                <Blueprint3.Icon
                  icon="document"
                  iconSize={20}
                >
                  <span
                    className="bp3-icon bp3-icon-document"
                    icon="document"
                  >
                    <svg
                      data-icon="document"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <desc>
                        document
                      </desc>
                      <path
                        d="M11.98 0h-8c-.55 0-1 .45-1 1v18c0 .55.45 1 1 1h13c.55 0 1-.45 1-1V6l-6-6zm4 18h-11V2h6v5h5v11z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </span>
                </Blueprint3.Icon>
                 
                Save
              </a>
            </SideNavstyle__Link>
            <SideNavstyle__Link
              href="javascript:void(0)"
              onClick={[Function]}
            >
              <a
                className="SideNavstyle__Link-sc-1s7cd9d-2 jBzRZo"
                href="javascript:void(0)"
                onClick={[Function]}
              >
                <Blueprint3.Icon
                  icon="document-open"
                  iconSize={20}
                >
                  <span
                    className="bp3-icon bp3-icon-document-open"
                    icon="document-open"
                  >
                    <svg
                      data-icon="document-open"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <desc>
                        document-open
                      </desc>
                      <path
                        d="M8 15c0 .55.45 1 1 1s1-.45 1-1v-5c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1h2.59L1.3 16.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L8 12.41V15zm5-15H5c-.55 0-1 .45-1 1v6h2V2h6v5h5v11H6v-.76L4.04 19.2c.1.45.48.8.96.8h13c.55 0 1-.45 1-1V6l-6-6z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </span>
                </Blueprint3.Icon>
                 
                open
              </a>
            </SideNavstyle__Link>
            <SideNavstyle__Link
              href="javascript:void(0)"
              onClick={[Function]}
            >
              <a
                className="SideNavstyle__Link-sc-1s7cd9d-2 jBzRZo"
                href="javascript:void(0)"
                onClick={[Function]}
              >
                <Blueprint3.Icon
                  icon="print"
                  iconSize={20}
                >
                  <span
                    className="bp3-icon bp3-icon-print"
                    icon="print"
                  >
                    <svg
                      data-icon="print"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <desc>
                        print
                      </desc>
                      <path
                        d="M14 16H6v-4H4v5c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-5h-2v4zm2-13c0-.55-.45-1-1-1H5c-.55 0-1 .45-1 1v1h12V3zm3 2H1c-.55 0-1 .45-1 1v7c0 .55.45 1 1 1h2v-3h14v3h2c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1zm-1 4h-2V7h2v2z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </span>
                </Blueprint3.Icon>
                 Print
              </a>
            </SideNavstyle__Link>
            <SideNavstyle__Link
              href="javascript:void(0)"
              onClick={[Function]}
            >
              <a
                className="SideNavstyle__Link-sc-1s7cd9d-2 jBzRZo"
                href="javascript:void(0)"
                onClick={[Function]}
              >
                <Blueprint3.Icon
                  icon="export"
                  iconSize={20}
                >
                  <span
                    className="bp3-icon bp3-icon-export"
                    icon="export"
                  >
                    <svg
                      data-icon="export"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <desc>
                        export
                      </desc>
                      <path
                        d="M5 7c.28 0 .53-.11.71-.29L9 3.41V15c0 .55.45 1 1 1s1-.45 1-1V3.41l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71l-5-5C10.53.11 10.28 0 10 0s-.53.11-.71.29l-5 5A1.003 1.003 0 005 7zm14 7c-.55 0-1 .45-1 1v3H2v-3c0-.55-.45-1-1-1s-1 .45-1 1v4c0 .55.45 1 1 1h18c.55 0 1-.45 1-1v-4c0-.55-.45-1-1-1z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </span>
                </Blueprint3.Icon>
                 
                Export to SVG
              </a>
            </SideNavstyle__Link>
            <SideNavstyle__Link
              href="javascript:void(0)"
            >
              <a
                className="SideNavstyle__Link-sc-1s7cd9d-2 jBzRZo"
                href="javascript:void(0)"
              >
                <Blueprint3.Icon
                  icon="presentation"
                  iconSize={20}
                >
                  <span
                    className="bp3-icon bp3-icon-presentation"
                    icon="presentation"
                  >
                    <svg
                      data-icon="presentation"
                      height={20}
                      viewBox="0 0 20 20"
                      width={20}
                    >
                      <desc>
                        presentation
                      </desc>
                      <path
                        d="M19 1h-8c0-.55-.45-1-1-1S9 .45 9 1H1c-.55 0-1 .45-1 1s.45 1 1 1h1v11c0 .55.45 1 1 1h4.59L4.3 18.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71L9 16.41V19c0 .55.45 1 1 1s1-.45 1-1v-2.59l3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L12.41 15H17c.55 0 1-.45 1-1V3h1c.55 0 1-.45 1-1s-.45-1-1-1zm-3 12H4V3h12v10z"
                        fillRule="evenodd"
                        key="0"
                      />
                    </svg>
                  </span>
                </Blueprint3.Icon>
                 
                Presentation Mode
              </a>
            </SideNavstyle__Link>
            <SideNavstyle__Footer>
              <div
                className="SideNavstyle__Footer-sc-1s7cd9d-4 iSChnb"
              >
                <Blueprint3.Divider>
                  <div
                    className="bp3-divider"
                  />
                </Blueprint3.Divider>
                <SideNavstyle__Link
                  href="https://github.com/shubham-thakare/BMC-Creator"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <a
                    className="SideNavstyle__Link-sc-1s7cd9d-2 jBzRZo"
                    href="https://github.com/shubham-thakare/BMC-Creator"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    View on GitHub
                     
                    <Blueprint3.Icon
                      icon="share"
                      iconSize={14}
                    >
                      <span
                        className="bp3-icon bp3-icon-share"
                        icon="share"
                      >
                        <svg
                          data-icon="share"
                          height={14}
                          viewBox="0 0 16 16"
                          width={14}
                        >
                          <desc>
                            share
                          </desc>
                          <path
                            d="M10.99 13.99h-9v-9h4.76l2-2H.99c-.55 0-1 .45-1 1v11c0 .55.45 1 1 1h11c.55 0 1-.45 1-1V7.24l-2 2v4.75zm4-14h-5c-.55 0-1 .45-1 1s.45 1 1 1h2.59L7.29 7.28a1 1 0 00-.3.71 1.003 1.003 0 001.71.71l5.29-5.29V6c0 .55.45 1 1 1s1-.45 1-1V1c0-.56-.45-1.01-1-1.01z"
                            fillRule="evenodd"
                            key="0"
                          />
                        </svg>
                      </span>
                    </Blueprint3.Icon>
                  </a>
                </SideNavstyle__Link>
              </div>
            </SideNavstyle__Footer>
          </div>
        </SideNavstyle__ToggleHide>
      </div>
    </SideNavstyle__SideNavbar>
  </SideNav>
</ContextProviderWrapper>
`;
